### PR TITLE
Bug/skale 4430 skaled restart policy not working

### PR DESCRIPTION
**Pull request by badrogger**
- Added restarting procedure in case schain is failed with non zero and 200 exit code 
- Added restarting procedure in case schain rpc check is not working

### CODE REVIEW

**Review by olehnikolaiev — DISMISSED** [review]

File: core/schains/checks.py [hunk 1]
```diff
@@ -94,30 +98,28 @@ def firewall_rules(self) -> bool:
...
 
     @property
     def rpc(self) -> bool:
         """Checks that local skaled RPC is accessible"""
+        res = False
         if self.config:
             http_endpoint = get_local_schain_http_endpoint(self.name)
-            return check_endpoint_alive(http_endpoint)
-        return False
+            res = check_endpoint_alive(http_endpoint)
```
Comment (dmytrotkk) [inline core/schains/checks.py:121]: what is the purpose of this change

File: tests/schains/checks_test.py [hunk 2]
```diff
@@ -135,17 +148,28 @@ def test_ima_check(sample_checks, sample_false_checks):
...
-        assert not sample_false_checks.rpc
+
+    for _ in range(4):
+        # r = SChainRecord.get_by_name(schain_name)
+        # assert r.failed_rpc_count == i
+        assert not sample_checks.rpc
+
+    with mock.patch('requests.post', new=request_mock(res_mock)):
+        assert sample_checks.rpc
+    # r = SChainRecord.get_by_name(schain_name)
```
Comment (dmytrotkk) [inline tests/schains/checks_test.py]: do we need these comments?

**Review by dmytrotkk — DISMISSED** [review]

File: core/schains/monitor.py [hunk 3]
```diff
@@ -73,7 +73,7 @@
 logger = logging.getLogger(__name__)
 
 CONTAINERS_DELAY = 20
-SCHAIN_MONITOR_SLEEP_INTERVAL = 500
+SCHAIN_MONITOR_SLEEP_INTERVAL = 30
```
Comment (dmytrotkk) [inline core/schains/monitor.py]: test value pushed

**Review by dmytrotkk — CHANGES_REQUESTED** [review]

**Review by dmytrotkk — APPROVED** [review]
